src/azure-cli/azure/cli/command_modules/vm/custom.py

```diff
@@ -2687,7 +2687,16 @@ def update_vmss(cmd, resource_group_name, name, license_type=None, no_wait=False
                 scale_in_policy=None, priority=None, max_price=None, proximity_placement_group=None,
                 enable_automatic_repairs=None, automatic_repairs_grace_period=None, **kwargs):
     vmss = kwargs['parameters']
-    client = _compute_client_factory(cmd.cli_ctx)
+    aux_subscriptions = None
+    if vmss and hasattr(vmss, 'virtual_machine_profile') and vmss.virtual_machine_profile and \
```

Comment:
why `hasattr(vmss, 'virtual_machine_profile')`?
Do we need `hasattr(vmss.virtual_machine_profile, 'storage_profile ')`?

Reply:
Very nice find. Because when `--instance-id` is specified, the model updated will be `VirtualMachineScaleSetVM` which has no `virtual_machine_profile`. Normally this command is updating `VirtualMachineScaleSet`. So `hasattr` here is to make sure the current model is `VirtualMachineScaleSet`. And there is no need to do this check for its child properties.